power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -598,17 +598,130 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
     static constexpr auto transformer_index_of = container_impl::get_cls_pos_v<T, TransformerTypes...>;
     static_assert(((transformer_index_of<TransformerTypes> < sizeof...(TransformerTypes)) && ...));
 
+    struct BinarySearch {
```

Comment:
i really like the idea of making this a generic search engine!

Reply:
XD Then we would again go ahead and make something not easily readable for others later...

On the other hand, this is by no means a standard way of doing binary search as it is highly tailored made to the tap changing strategies. Not sure how much effort that would be to make it generic. Could be an option when we face another similar task.